accounts-db/src/accounts_db.rs

```diff
@@ -8868,7 +8868,9 @@ impl AccountsDb {
                             );
                             accessor.check_and_get_loaded_account(|loaded_account| {
                                 let data_len = loaded_account.data_len();
-                                accounts_data_len_from_duplicates += data_len;
+                                if loaded_account.lamports() > 0 {
+                                    accounts_data_len_from_duplicates += data_len;
```

Comment:
i think this just affects metrics? Does this bite us anywhere else? Should we update rewrites to detect this and get rid of the data?

Reply:
Yes, I tracked down the usage of `accounts_data_len`. It is only used in metrics.

Reply:
Does this bite us anywhere else? Should we update rewrites to detect this and get rid of the data?
We should track down why this zero lamport account is not getting thrown away. Is it being rewritten with rewrites?

Reply:
>  Should we update rewrites to detect this and get rid of the data?

Currently rewrite won't visit zero lamport accounts. That's probably why this accounts is here forever. 

Here is the PR that allows us to visit non empty zero lamport accounts and rewrite it with empty data.

https://github.com/anza-xyz/agave/pull/3290